src/resilience/exec_space/openMP/OpenMPResSubscriber.hpp

```diff
@@ -198,57 +457,90 @@ struct ResilientDuplicatesSubscriber {
     }
 
     auto &res = *static_cast< CombineDuplicates<View> * >( pos->second.get());
-
     // If inserted in the cache map then create copies and reinitialize
-    if (inserted) {
+
+//TODO: There may be some subtle issues with this in DMR since it only checks copy[0] and it won't trigger on the second copy, which would still need resizing. Consider a solution
+    bool extents_resized = false;
+    for (size_t i = 0; i < View::rank; ++i)
+    {   
+      if (original.extent(i) != res.copy[0].extent(i) )
+      {   
+        extents_resized = true;
+        break;
+      }   
+    } 
+
+#if 0    
+#ifdef KR_ENABLE_DMR
+
+     // DMR variant initialization of copies
+     // only 1 initially, 2 on fail
+     if (dmr_failover_to_tmr && res.copy[1].data()==nullptr){
+      // Create second copy
+        set_duplicate_view(res.copy[1], original, 1);
+      }
+#endif
+#endif
+
+    if (inserted || extents_resized) {
       res.original = original;
 
+#ifdef KR_ENABLE_DMR
+      if (dmr_failover_to_tmr){
+        // Create second copy
+        set_duplicate_view(res.copy[1], original, 1);
+      }else{
+
+	      //TODO: for dmr want this to be in main inserted if
+	      //and uncomment if 0
+      // only 1 initially, 2 on fail
+      // Create first copy for DMR
+      set_duplicate_view(res.copy[0], original, 0);
+      }
+#else
+
+      // Reinitialize self to be like other (same dimensions, etc)
       for (int i = 0; i < 2; ++i) {
           set_duplicate_view(res.copy[i], original, i);
       }
+
+#endif
```

Comment:
Something like this?
```suggestion
  for(int copy = 0; copy < res.duplicate_count; copy++){
    for(int dim = 0; dim < View::rank; dim++){
      if(original.extent(dim) != res.copy[copy].extent(dim)){
        set_duplicate_view(res.copy[copy], original, copy);
        break;
      }
    }
  }
```

Reply:
Actually I don't think this would fix the DMR issue, since the duplicate will be 1 until after this function is called.